### PR TITLE
Adding Elysian

### PR DESCRIPTION
**Pull request by infinityis**
adding Elysian

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/custommk/elysian/config.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#define EXTERNAL_EEPROM_SPI_SLAVE_SELECT_PIN B7
+#define EXTERNAL_EEPROM_SPI_CLOCK_DIVISOR 4
+
+// SPI configuration
+#define SPI_DRIVER SPID1
+#define SPI_SCK_PIN B3
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+
+#define DEBOUNCE 10
```
Comment (zvecr) [inline keyboards/custommk/elysian/config.h:16]:
```suggestion

```

File: keyboards/custommk/elysian/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+    "manufacturer": "customMK",
+    "keyboard_name": "Elysian",
+    "maintainer": "customMK",
+    "bootloader": "stm32-dfu",
```
Comment (zvecr) [inline keyboards/custommk/elysian/keyboard.json:5]:
```suggestion
    "bootloader": "stm32-dfu",
    "debounce": 10,
```

File: keyboards/custommk/elysian/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 customMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC, MO(1),
```
Comment (zvecr) [inline keyboards/custommk/elysian/keymaps/default/keymap.c:8]: `MO(1)` here would go out of bounds.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/custommk/elysian/config.h (hunk 1 above)
Comment (infinityis) [reply keyboards/custommk/elysian/config.h]: Moved debounce to data driven, thank you!

File: keyboards/custommk/elysian/keyboard.json (hunk 2 above)
Comment (infinityis) [reply keyboards/custommk/elysian/keyboard.json:5]: Moved debounce to data driven, thank you!

File: keyboards/custommk/elysian/keymaps/default/keymap.c (hunk 3 above)
Comment (infinityis) [reply keyboards/custommk/elysian/keymaps/default/keymap.c]: Changed default keycode assignment to avoid going out of bounds, thank you!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**